pysb/simulator/cuda_ssa.py

```diff
@@ -0,0 +1,254 @@
+from __future__ import print_function
+
+#
+try:
+    import pycuda
+    import pycuda.autoinit
+    import pycuda as cuda
+    import pycuda.compiler
+    import pycuda.tools as tools
+    import pycuda.driver as driver
+    import pycuda.gpuarray as gpuarray
+except ImportError:
+    pycuda = None
+
+import numpy as np
+import os
+import time
+from pysb.logging import setup_logger
+import logging
+from pysb.pathfinder import get_path
+from pysb.simulator.base import SimulationResult, SimulatorException
+from pysb.simulator.ssa_base import SSABase
+
+
+class CUDASimulator(SSABase):
+    """
+    CUDA simulator for NVIDIA gpus
+
+    Requires pycuda, cuda, and cuda compatible NVIDIA gpus.
+
+    Parameters
+    ----------
+    model : pysb.Model
+        Model to simulate.
+    tspan : vector-like, optional
+        Time values over which to simulate. The first and last values define
+        the time range. Returned trajectories are sampled at every value unless
+        the simulation is interrupted for some reason, e.g., due to
+        satisfaction
+        of a logical stopping criterion (see 'tout' below).
+    initials : vector-like or dict, optional
+        Values to use for the initial condition of all species. Ordering is
+        determined by the order of model.species. If not specified, initial
+        conditions will be taken from model.initial_conditions (with
+        initial condition parameter values taken from `param_values` if
+        specified).
+    param_values : vector-like or dict, optional
+        Values to use for every parameter in the model. Ordering is
+        determined by the order of model.parameters.
+        If passed as a dictionary, keys must be parameter names.
+        If not specified, parameter values will be taken directly from
+        model.parameters.
+    verbose : bool, optional (default: False)
+        Verbose output.
+
+    Attributes
+    ----------
+    verbose: bool
+        Verbosity flag passed to the constructor.
+    model : pysb.Model
+        Model passed to the constructor.
+    tspan : vector-like
+        Time values passed to the constructor.
+    """
+    _supports = {'multi_initials': True, 'multi_param_values': True}
+
+    def __init__(self, model, verbose=False, tspan=None, **kwargs):
+        if pycuda is None:
+            raise SimulatorException('pycuda library required for {}'
```

Comment:
Maybe an`ImportError` would be better here?

Reply:
Fixed